lib/src/components/stepper_input/stepper_input.dart

```diff
@@ -0,0 +1,182 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+import '../../../zeta_flutter.dart';
+
+/// A stepper input, also called numeric stepper, is a common UI element that allows uers to input a number or value simply by clicking the plus and minus buttons.
+class ZetaStepperInput extends StatefulWidget {
+  /// Creates a new [ZetaStepperInput]
+  const ZetaStepperInput({
+    this.rounded = true,
+    this.size = ZetaWidgetSize.medium,
+    this.initialValue,
+    this.min,
+    this.max,
+    this.onChange,
+    super.key,
+  }) : assert(
+          (min == null || (initialValue ?? 0) >= min) && (max == null || (initialValue ?? 0) <= max),
+          'Initial value must be inside given min and max values',
+        );
+
+  /// {@macro zeta-component-rounded}
+  final bool rounded;
+
+  /// The size of the stepper input.
+  final ZetaWidgetSize size;
```

Comment:
I don't think we should use this enum if we aren't using all the values. Maybe we need an enum with just 2 sizes? or even just a boolean `isLarge`?